### PR TITLE
MySGonks page logic implementation 

### PR DESCRIPTION
**Pull request by lin8231**
* LongSGonksList is now being fed an array of investments which it should display
    * Filters for unsold sgonks (filters using dateSoldMilliSeconds == 0)
* LongSGonksList dynamically renders LongSGonkRows, each is passed one investment object from the investments array
* Chart doesn't yet exist on this page, to be implemented in a "Add Charts to pages" PR later 
* Buttons currently don't do anything (the pages they point to don't yet exist) - to be done last in a "Inner page routing" PR
* Communication is one way only (backend -> frontend) at the moment, selling functionality not yet there 

This is branched off **fe-mysgonks-subcomponents** which is branched off **fe-mysgonks-ui** which is branched off **fe-link-to-servlets** the PR isn't actually that big it's only the last... 7 commits

### CODE REVIEW

**Review by Phoebe-Khokgawe — APPROVED** [review]

File: project/frontend/src/components/SGonksLists/ShortSGonksList/ShortSGonksRow/ShortSGonksRow.js [hunk 1]
```diff
@@ -2,13 +2,12 @@ import React from "react";
 import classes from "./ShortSGonksRow.module.css";
 
 const ShortSGonksRow = (props) => {
+  console.log(props);
```
Comment (texm) [inline project/frontend/src/components/SGonksLists/ShortSGonksList/ShortSGonksRow/ShortSGonksRow.js:5]: intentional log?

File: project/frontend/src/containers/Dashboard/Dashboard.js [hunk 2]
```diff
@@ -6,107 +6,43 @@ import Block from "../../components/UI/Block/Block";
...
-      currentPrice: "321809",
-      priceYesterday: "321321",
-    },
-    {
-      searchTerm: "eek",
-      currentPrice: "321",
-      priceYesterday: "321521",
-    },
-  ];
+  console.log(props);
```
Comment (texm) [inline project/frontend/src/containers/Dashboard/Dashboard.js:14]: intentional log?

File: project/frontend/src/containers/MySGonks/MySGonks.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import Block from "../../components/UI/Block/Block";
+import LinkButton from "../../components/UI/LinkButton/LinkButton";
+import LongSGonksList from "../../components/SGonksLists/LongSGonksList/LongSGonksList";
+
+const MySGonks = (props) => {
+  console.log(props);
+  const unsoldInvestments = props.investments.filter(
+    (investment) => investment.dateSoldMilliSeconds == 0
+  );
+  console.log(unsoldInvestments);
```
Comment (texm) [inline project/frontend/src/containers/MySGonks/MySGonks.js:12]: 2 logs

File: project/frontend/src/App.js [hunk 4]
```diff
@@ -60,11 +101,108 @@ function App() {
...
+            //   },
+            //   {
+            //     amtInvested: 213,
+            //     currentValue: 321,
+            //     dataPoints: [162, 152, 150, 1321, 13213],
+            //     dateInvestedMilliSeconds: 1611705600000,
+            //     dateSoldMilliSeconds: 0,
+            //     searchItem: "fadfsdafsdfdas",
+            //   },
+            // ]}
```
Comment (texm) [inline project/frontend/src/App.js:200]: prefer multiline comment

File: project/frontend/src/components/SGonksLists/ShortSGonksList/ShortSGonksRow/ShortSGonksRow.js [hunk 5]
```diff
@@ -2,13 +2,12 @@ import React from "react";
...
-    const diff =
-      props.sgonk.currentPrice > props.sgonk.priceYesterday
-        ? "+"
-        : props.sgonk.currentPrice < props.sgonk.priceYesterday
-        ? "-"
-        : "=";
+    const priceDelta =
+      props.datapoints[props.datapoints.length - 1] -
+      props.datapoints[props.datapoints.length - 2];
+    const diff = priceDelta > 0 ? "+" : priceDelta < 0 ? "-" : "=";
```
Comment (texm) [inline project/frontend/src/components/SGonksLists/ShortSGonksList/ShortSGonksRow/ShortSGonksRow.js:10]: nit for readability: complex ternaries often make things harder to read. I'd suggest adding some brackets & newlines to help here, eg
```js
const diff = (priceDelta > 0) ? "+" 
           : (priceDelta < 0) ? "-" 
           : "=";
```

**Review by texm — APPROVED** [review]
Nice work, LGTM broadly. some comments to address before merging

**Comment by roonil-wazlib** [comment]
@googlebot I consent.

File: project/frontend/src/App.js (hunk 4 above)
Comment (lin8231) [reply project/frontend/src/App.js]: Done

File: project/frontend/src/containers/Dashboard/Dashboard.js (hunk 2 above)
Comment (lin8231) [reply project/frontend/src/containers/Dashboard/Dashboard.js]: Done

File: project/frontend/src/containers/MySGonks/MySGonks.js (hunk 3 above)
Comment (lin8231) [reply project/frontend/src/containers/MySGonks/MySGonks.js]: Done

File: project/frontend/src/components/SGonksLists/ShortSGonksList/ShortSGonksRow/ShortSGonksRow.js (hunk 5 above)
Comment (lin8231) [reply project/frontend/src/components/SGonksLists/ShortSGonksList/ShortSGonksRow/ShortSGonksRow.js:9]: Linter says no :(

**Pull request merged**